url11fb0973cf_raichuman9_Swimmer526_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/ubc/swim/world/characters/HumanChar.java

```diff
@@ -0,0 +1,387 @@
+import org.jbox2d.dynamics.joints.RevoluteJoint;
+import org.jbox2d.dynamics.joints.RevoluteJointDef;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ubc.swim.gui.SwimSettings;
+import ubc.swim.world.motors.GaussianTorqueMotor;
+import ubc.swim.world.motors.TorqueMotor;
+
+/**
+ * A roughly humanoid swimmer that uses directly optimized torque motors
+ * @author Ben Humberston
+ *
+ */
+public class HumanChar extends SwimCharacter {
+	private static final Logger log = LoggerFactory.getLogger(HumanChar.class);
+	
+	protected static final int NUM_GAUSSIANS_PER_MOTOR = 2;
+	protected static final int NUM_PARAMS_PER_GAUSSIAN = 3;
+	protected static final int NUM_PARAMS_PER_MOTOR = 1 + NUM_PARAMS_PER_GAUSSIAN * NUM_GAUSSIANS_PER_MOTOR; //+1 for period value
+	
+	//Hard-coded, but data-driven solution is too time consuming
+	//1 motor for head, 2 for arms, 2 for legs (arm & leg controls are mirrored between left & right sides)
+	protected static final int NUM_CONTROL_DIMENSIONS = 5 * (1 + (NUM_PARAMS_PER_GAUSSIAN * NUM_GAUSSIANS_PER_MOTOR));
+	
+	protected static final int MAX_STROKE_PERIOD = 5; //seconds
+	protected static final float MAX_DEFAULT_TORQUE = 50; //N-m
+	
+	//Body params
+	protected float height = 2; //2 meters
+	protected float headHeight = height / 8;
+	protected float headWidth = headHeight * 0.85f;
+	protected float armLen = height * 0.45f;
+	protected float upperArmLen = armLen * (1 / 2.2f);
+	protected float upperArmWidth = height / 16;
+	protected float lowerArmLen = armLen - upperArmLen;
+	protected float lowerArmWidth = upperArmWidth * 0.85f;
+	protected float legLen = height * 0.54f;
+	protected float upperLegLen = legLen * 0.46f;
+	protected float upperLegWidth = height / 10;
+	protected float lowerLegLen = legLen - upperLegLen;
+	protected float lowerLegWidth = upperLegWidth * 0.7f;
+	protected float footLen = height * 0.035f;
+	
+	protected Stroke stroke;
+	
+	protected ArrayList<Body> rightBodies;
+	protected ArrayList<Body> leftBodies;
+	
+	protected ArrayList<Joint> joints;
+	protected ArrayList<Joint> leftJoints;
+	protected ArrayList<Joint> rightJoints;
+	
+	//Motors for arms and legs, split by left and right sides
+	protected ArrayList<TorqueMotor> rightMotors;
+	protected ArrayList<TorqueMotor> leftMotors;
+	
+	protected float prevTorque = 0.0f;
+	
+	/**
+	 * Create a new human character with given stroke as goal
+	 * @param stroke
+	 */
+	public HumanChar(Stroke stroke) {
+		super();
+		
+		this.stroke = stroke;
+	
+		rightBodies = new ArrayList<Body>();
+		leftBodies = new ArrayList<Body>();
+		
+		joints = new ArrayList<Joint>();
+		
+		leftJoints = new ArrayList<Joint>();
+		rightJoints = new ArrayList<Joint>();
+		
+		rightMotors = new ArrayList<TorqueMotor>();
+		leftMotors = new ArrayList<TorqueMotor>();
+	}
+
+	@Override
+	public int getNumControlDimensions() { 
+		return NUM_CONTROL_DIMENSIONS;
+	}
+	
+	@Override
+	public void initialize(World world) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 109 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 284-289 into a separate method.